lgc/builder/BuilderImpl.cpp

```diff
@@ -559,9 +640,13 @@ Instruction *BuilderImpl::createWaterfallLoop(Instruction *nonUniformInst, Array
     if (nonUniformIndex)
       nonUniformIndices.push_back(nonUniformIndex);
   }
+
   if (nonUniformIndices.empty())
     return nonUniformInst;
 
+  if (nonUniformInst->getType()->isVoidTy())
+    scalarizeDescriptorLoads = false;
```

Comment:
Why this?

Reply:
I am sorry I missed this. This is for the case that we have the following intrinsic: @llvm.amdgcn.struct.buffer.store.format.v4f32()

Reply:
Can you please comment on that?

Reply:
Done

Reply:
This still doesn't make sense though. The "scalarization" is a function of the *descriptor inputs* to the intrinsic, not a function of its output. This should be removed.

Reply:
I remove it and I added two tests : scalarizationOfDescriptorLoadTest1.ll and scalarizationOfDescriptorLoadTest2.ll